message_ix_models/tools/costs/regional_differentiation.py

```diff
@@ -49,49 +49,49 @@ def get_weo_data() -> pd.DataFrame:
     # their respective sheet in the Excel file,
     # and the start row
     DICT_TECH_ROWS = {
-        "bioenergy_ccus": ["Renewables", 95],
-        "bioenergy_cofiring": ["Renewables", 75],
-        "bioenergy_large": ["Renewables", 65],
-        "bioenergy_medium_chp": ["Renewables", 85],
-        "ccgt": ["Gas", 5],
-        "ccgt_ccs": ["Fossil fuels equipped with CCUS", 25],
-        "ccgt_chp": ["Gas", 25],
-        "csp": ["Renewables", 105],
-        "fuel_cell": ["Gas", 35],
-        "gas_turbine": ["Gas", 15],
-        "geothermal": ["Renewables", 115],
-        "hydropower_large": ["Renewables", 45],
-        "hydropower_small": ["Renewables", 55],
-        "igcc": ["Coal", 35],
-        "igcc_ccs": ["Fossil fuels equipped with CCUS", 15],
-        "marine": ["Renewables", 125],
-        "nuclear": ["Nuclear", 5],
-        "pulverized_coal_ccs": ["Fossil fuels equipped with CCUS", 5],
-        "solarpv_buildings": ["Renewables", 15],
-        "solarpv_large": ["Renewables", 5],
-        "steam_coal_subcritical": ["Coal", 5],
-        "steam_coal_supercritical": ["Coal", 15],
-        "steam_coal_ultrasupercritical": ["Coal", 25],
-        "wind_offshore": ["Renewables", 35],
-        "wind_onshore": ["Renewables", 25],
+        "bioenergy_ccus": ["Renewables", 99],
+        "bioenergy_cofiring": ["Renewables", 79],
+        "bioenergy_large": ["Renewables", 69],
+        "bioenergy_medium_chp": ["Renewables", 89],
+        "ccgt": ["Gas", 9],
+        "ccgt_ccs": ["Fossil fuels equipped with CCUS", 29],
+        "ccgt_chp": ["Gas", 29],
+        "csp": ["Renewables", 109],
+        "fuel_cell": ["Gas", 39],
+        "gas_turbine": ["Gas", 19],
+        "geothermal": ["Renewables", 119],
+        "hydropower_large": ["Renewables", 49],
+        "hydropower_small": ["Renewables", 59],
+        "igcc": ["Coal", 39],
+        "igcc_ccs": ["Fossil fuels equipped with CCUS", 19],
+        "marine": ["Renewables", 129],
+        "nuclear": ["Nuclear", 9],
+        "pulverized_coal_ccs": ["Fossil fuels equipped with CCUS", 9],
+        "solarpv_buildings": ["Renewables", 19],
+        "solarpv_large": ["Renewables", 9],
+        "steam_coal_subcritical": ["Coal", 9],
+        "steam_coal_supercritical": ["Coal", 19],
+        "steam_coal_ultrasupercritical": ["Coal", 29],
+        "wind_offshore": ["Renewables", 39],
+        "wind_onshore": ["Renewables", 29],
     }
 
     # Dict of cost types to read in and the required columns
     DICT_COST_COLS = {"inv_cost": "A,B:D", "fix_cost": "A,F:H"}
 
     # Set file path for raw IEA WEO cost data
     file_path = package_data_path(
-        "iea", "WEO_2022_PG_Assumptions_STEPSandNZE_Scenario.xlsb"
+        "iea", "WEO_2023_PG_Assumptions_STEPSandNZE_Scenario.xlsx"
     )
 
     # Retrieve conversion factor
-    conversion_factor = registry("1.0 USD_2021").to("USD_2005").magnitude  # noqa: F841
+    conversion_factor = registry("1.0 USD_2022").to("USD_2005").magnitude  # noqa: F841
```

Comment:
```suggestion
    conversion_factor = registry("1.0 USD_2022").to("USD_2005").magnitude # if variable is needed later on
    _ = registry("1.0 USD_2022").to("USD_2005").magnitude # if not
```

Reply:
Thanks @glatterf42 ! The `conversion_factor` is an interesting case where it _is_ used later on here:

https://github.com/iiasa/message-ix-models/blob/0b0ee50c11d1a3c7dfa3b8dfe94afd9afbf35973/message_ix_models/tools/costs/regional_differentiation.py#L125

But when used inside the string, the linting does not pick it up as being used. Without the `# noqa: F841`, when I save the file, the linter will automatically delete the `conversion_factor` variable line. So that's why the `# noqa: F841` is put there. Alternatively, I can change the variable name to `_conversion_factor` and use it in the string as `@_conversion_factor`, and I think that should work as well. 😄

Reply:
Huh, I didn't know about such a use-case. However, reading up on it, I found [this note](https://pandas.pydata.org/pandas-docs/stable/user_guide/enhancingperf.html#expression-evaluation-via-eval): `DataFrame.eval()` is only recommended for `DataFrames` with more than 1e5 rows. Do you have any estimate about the number of rows in your `df`? If it's lower for sure, [`df.assign()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.assign.html#pandas.DataFrame.assign) might be more performant.

Reply:
I've changed the line to use `df.assign()` instead, so now `conversion_factor` is explicitly used :)

https://github.com/iiasa/message-ix-models/blob/5b3a1f820579562f0b8ac2a0e7e5368dbff8aefb/message_ix_models/tools/costs/regional_differentiation.py#L125

However, I will note that I believe encounter this same use-case in other places (especially when I use `df.query()`).